### ISSUE TITLE
[Meta] Sync open tabs

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#5762]
### User Story
As a user, I want to be able to access the tabs that I have open on other Firefox connected devices in Fenix and vice versa, so I can have task continuity across my devices and easily access the information I need regardless of the platform.

Dependency:
Dependency on Sync team and A-C
- Expose [session-history](https://github.com/mozilla-mobile/android-components/issues/1340)
[A-C board](https://github.com/orgs/mozilla-mobile/projects/11) 

### Acceptance Criteria
- [x] I can choose to sync my tabs across my Fx connected devices
- [x] I can access my synced tabs in other connected devices on Fenix
- [x] I can access my synced Fenix tabs on my other Fx connected devices

**Comment by linacambridge** [comment]
Related: #4479

**Comment by lime124** [comment]
this requires a-s that should be completed in dec.

this could be a good early on ux and eng collaboration.

**Comment by lime124** [comment]
followed up with grisha - some of this work can be done in parallel to the accounts team.  e.g. we need a synced-tabs API from a-s, but it doesn't necessarily need to sync yet - or even exist - to do some of the the fenix integration work (library UI, etc)

**Comment by vesta0** [comment]
@lime124 can we re-use Fennec UX for this?

**Comment by yoasif** [comment]
I think at this point, it would be a mistake to reuse Fennec UX here. I recently commented https://github.com/mozilla-mobile/fenix/issues/2486#issuecomment-578003913 where I noted that Fennec's reopening closed tabs UI isn't as good as the one in Vivaldi since it requires users to move away from where they manage tabs to a separate history area.

Here, the situation is even worse, since synced tabs don't appear even in the overburdened (imo) Library area on Firefox desktop. Having it appear in History in Fennec feels like a mistake (but understandable in a sense, since I had to have looked at those tabs in the past [history] for it to be on my sync list). 

I think the biggest issue with Library in general is that unlike Fennec, it is not a tab or page on the main page, but rather hidden behind a menu, which the Vivaldi tab switcher solves with an icon in the tab switcher that opens a tab with your synced tabs.

Image here:

![image](https://user-images.githubusercontent.com/152493/73052556-64a58c00-3e53-11ea-81eb-e6362fe00483.png)

Also, FWIW, I guess three years ago, synced tabs appeared as a "tab" in the home on Fennec - 

![image](https://user-images.githubusercontent.com/152493/73053244-3aed6480-3e55-11ea-8546-076e32bf00b9.png)

I tried installing Firefox 47 on my Pixel 2, but it just crashes when I launch it, unfortunately. 

Obviously this comment is *just* about how to get to the synced tabs, not the synced tab display itself - which I think is fine in Fennec.

**Comment by vesta0** [comment]
Thanks for your input @yoasif after looking into it more carefully I agree with you that we shouldn't re-use Fennec UI for this. We will look into designing something that is intuitive and easy to find.

**Comment by madb1lly** [comment]
I now use Fenix Nightly as my daily driver and this is a feature I miss.

**Comment by yoasif** [comment]
Beta users are missing this feature: https://www.reddit.com/r/firefox/comments/g18t9x/firefox_previewbeta_on_android_and_synced_tabs/

**Comment by madb1lly** [comment]
Since I didn't know that Fenix doesn't back up settings to Google nor via ADB I actually lost my whole session, tabs, collections etc. If there was sync tabs then I wouldn't have lost this.

We need sync tabs not only for productivity/day-to-day benefits but also for backup - ultimately this is security, because availability is one of the three pillars of security.

Cheers 🙂

**Comment by yoasif** [comment]
This might work well in tab tray if tab tray were to be tabbed. See also https://github.com/mozilla-mobile/fenix/issues/7913

**Comment by liuche** [comment]
Hi @topotropic! I had been looking at the UX schedule (that Tif put in the kanban) and it looked like this would be ready soon, and we have an intern who wanted to work on Synced Tabs. He'll be getting started with good first bugs for a few days, and then we'll be ready to have him work on this - probably starting on the backend, fetching/syncing tabs, that sort of thing. Let me know how that sounds!

**Comment by vesta0** [comment]
After talking to @liuche I am moving this to `Ready for QA` since the acceptance criteria have been met. @topotropic please create a new issue for the post-launch design changes.

**Comment by liuche** [comment]
We're waiting for the fix for #11294 before unflagging it, but AS knows that this is important for release.

**Comment by liuche** [comment]
Okay, the crashfix is ready, and we are ready to turn off the feature flag.

**Comment by Diana-Rus** [comment]
Hi verified with Nightly 6/26 with Google Pixel 3XL (Android 9), Samsung Galaxy S9 (Android 8) and for desktop Windows 10 - Firefox Nightly 79.0a1 (2020-06-26) (64-bit) 

**1. Sync my tabs across my Fx connected devices -> phone to desktop**
1.1 Automatic sync when logging with fx account on phone, while connected to desktop ✔️
1.2 Automatic sync - adding a new tab on phone and access Synced Tabs ✔️
**2. Sync my tabs across my Fx connected devices -> desktop to phone**
2.1 Automatic sync when logging with fx account on desktop, while connected to phone ✔️
2.2 Automatic sync - adding a new tab on desktop and access Synced Tabs ✔️
2.3 Manually triggered sync - new tabs added on desktop are present on mobile✔️
**3 Sync my tabs across my Fx connected devices -> phone to phone**
3.1 Automatic sync - adding a new tab on phone 2(while logged) and then accessing Synced tabs for phone 1-device and tabs are present. ✔️
3.2 Update an already sync tab with a different domain on phone 2 -> manually trigger the sync on phone 2 -> manually trigger the sync on phone 1- the tab with the updated domain is present on the phone 1 Synced tabs list ❌ Result: the tab's position is not updated with the new domain on the phone 1 Synced tabs list
3.3 Delete tab from phone 2 -> manually trigger the sync on phone 2 -> manually trigger the sync on phone 1 - as a result the tab is not present on in Synced tabs on phone 1 ❌ Result: the removed tab from phone 2 is present in Synced tabs on phone 1
4. I can access my synced tabs in other connected devices on Fenix ✔️
5. I can access my synced Fenix tabs on my other Fx connected devices ✔️
6. Sync tabs across browser apps - R-B ✔️

Should the user be left to do manual sync in Synced tabs, how should it work for this 2 cases: 
- When checking the sync from desktop to phone while in Sync tabs, add a new tab on desktop (the new tab from desktop is not present on mobile)❓ 
- While being logged with fx account on phone 2, while connected with fx account to phone 1 (phone 2 will not be present in Synced tabs on phone 1)❓
- Should Fennec be taken into consideration in having to be displayed in Synced tabs(Fenix)? At the moment Fennec it's not displayed in the Synced tabs list

**Comment by liuche** [comment]
It looks like we might not be refreshing the Synced Tabs view in response to immediate syncs.

- new tabs not showing up immediately in Synced Tabs from other devices
- newly synced devices (or Fennec devices) don't always show up

**Comment by vesta0** [comment]
This work is completed. Individual bugs are tracked separately.
